### ISSUE TITLE
Add ignore attribute on an index

### ISSUE THREAD

**Issue by generalpiston** [prisma/prisma#13719]
## Problem

Currently, if you add an index to a table without a reference in the model, it will be marked for removal in a following migration. Some indices are out of scope of the index (ie. an index with an expression).

## Suggested solution

Adding an ignore attribute on an index would allow db admins to ignore a specific index and manage it outside of the model.

## Alternatives

`@@ignore` on an index... however attributes work on fields and models.

## Additional context

Currently, when I try to create a new migration while a custom index exists, I get the following message:

```
[*] Changed the `...` table
  [+] Added index on columns (...)

✔ We need to reset the PostgreSQL database "..." at "localhost:5432".
Do you want to continue? All data will be lost. …
```

Also, the generated migration will have a query to drop the index.

**Comment by janpio** [comment]
Would adding support for indexes with expressions be the better solution here? Please search our issues to make sure the type of index this applies to is adequately represented with a feature request already.

**Comment by generalpiston** [comment]
I think this feature could act as a stop gap measure when Prisma is missing a feature for indices in general.

**Comment by AaronFriel** [comment]
@janpio I suspect no. There will almost always be gaps or new features between what Prisma supports and what the latest SQL database engines support. A great examples is:
- #18442

Where even though the Vector type can be used via `Unsupported`, indexes cannot be maintained in migrations because the `ivfflat` index type and `WITH (lists = ...)` clause is not supported.

**Comment by IchiharaGaku** [comment]
@janpio 

I am using pgvector as an extension with Prisma. I want to utilize the HNSW index with pgvector version 0.5.1. Since such an implementation was not supported in the schema, I manually created a migration file and wrote the create index command.

```sql
CREATE INDEX ON "Vector" USING hnsw (embedding vector_cosine_ops);
```

However, when I attempt to migrate, the shadow database outputs an error because it does not recognize the unsupported type for the embedding column as having dimensions.

```yaml
npx prisma migrate dev
Environment variables loaded from .env
Prisma schema loaded from prisma/schema.prisma
Datasource "db": PostgreSQL database "sample", schema "public" at "localhost:5432"

Error: P3006

Migration `20231107144953_hogehoge` failed to apply cleanly to the shadow database.
Error:
ERROR: column does not have dimensions
   0: schema_core::state::DevDiagnostic
             at schema-engine/core/src/state.rs:270
```

Using HNSW is urgent, and I would like to skip the checks done by the shadow database.
Is there any good way to do this?

**Comment by janpio** [comment]
No. Please open a new issue so we can figure out why the shadow database does think that this column does not have dimensions.

**Comment by prashantbhudwal** [comment]
I would recommend using OpenAI assistants API with built-in RAG. There is no need for a vector store anymore. :)
